packages/core/src/stylesheet.ts

```diff
@@ -68,8 +68,24 @@ export class Stylesheet {
     this.processCompoundVariants(config)
   }
 
-  processAtomicSlotRecipe = (recipe: Pick<SlotRecipeConfig, 'base' | 'variants' | 'compoundVariants'>) => {
+  processAtomicSlotRecipe = (
+    recipe: Pick<SlotRecipeConfig, 'base' | 'variants' | 'compoundVariants'> & Partial<Pick<SlotRecipeConfig, 'slots'>>,
+  ) => {
+    if (!recipe.slots) {
+      const slots = new Set<string>()
+      Object.keys(recipe.base ?? {}).forEach((name) => {
+        slots.add(name)
+      })
+
+      Object.keys(recipe.variants ?? {}).forEach((name) => {
```

Comment:
This might be a bug since recipe.variants represents each variant key, not slot key.

we probably need to get the keys of each variant to properly compute the slots. I’d recommend refactoring this to a “inferSlots” function.